irods_browser_app/views.py

```diff
@@ -0,0 +1,137 @@
+from irods.session import iRODSSession
+from irods.exception import CollectionDoesNotExist
+from hs_core import hydroshare
+
+irods_sess = None
+
+def search_ds(coll):
+    store = {}
+    file = []
+    folder = []
+    if coll.data_objects:
+        for files in coll.data_objects:
+            file.append(files.name)
+    if coll.subcollections:
+        for folders in coll.subcollections:
+            folder.append(folders.name)
+
+    store['files'] = file
+    store['folder'] = folder
+    return store
+
+# Create your views here.
+def login(request):
+    if request.method == 'POST':
+        port = int(request.POST['port'])
+        user = str(request.POST['user'])
+        password = str(request.POST['password'])
+        zone = str(request.POST['zone'])
+        host = str(request.POST['host'])
+        datastore = "/%s/home/%s" % (zone, user)
+
+        response_data = {}
+
+        global irods_sess
+        irods_sess = iRODSSession(user=user, password=password, zone=zone, host=host, port=port)
+
+        try:
+            irods_sess.collections.get(datastore)
+        except CollectionDoesNotExist:
+            request.session['irods_loggedin'] = False
+            request.session['login_message'] = 'iRODS login failed'
+            request.session['irods_file_name'] = ''
+            response_data['irods_loggedin'] = False
+            response_data['login_message'] = 'iRODS login failed'
+            response_data['irods_file_name'] = ''
+            response_data['error'] = "iRODS collection does not exist"
+            return HttpResponse(
+                json.dumps(response_data),
+                content_type="application/json"
+            )
+        else:
+            request.session["user"] = user
+            request.session["datastore"] = datastore
+            request.session["password"] = password
+            request.session["port"] = port
+            request.session["host"] = host
+            request.session["zone"] = zone
+            request.session['irods_loggedin'] = True
+            request.session['irods_file_name'] = ''
+
+            response_data['user'] = user
+            response_data['datastore'] = datastore
+            response_data['irods_loggedin'] = True
+            response_data['irods_file_name'] = ''
+            return HttpResponse(
+                json.dumps(response_data),
+                content_type = "application/json"
+            )
+    else:
+        return HttpResponse(
+            json.dumps({"error": "Not POST request"}),
+            content_type="application/json"
+        )
+
+def store(request):
+    return_object = []
+    global irods_sess
+    if not irods_sess:
+        irods_sess = iRODSSession(user=request.session.get('user'), password=request.session.get('password'),
+                                  zone=request.session.get('zone'), host=request.session.get('host'),
+                                  port=request.session.get('port'))
```

Comment:
I am not experienced with handling multiple processes in the web world at all, but I get your point. In terms of keeping credentials in the browser to submit with each request, we cannot do this for password, right? I will try to do this and try to avoid use of global session object, but not sure how to handle password since we cannot even pass them around between request, right? Thanks for any specific suggestions/ideas.